### PR TITLE
feat(docker): official Docker image

### PR DESCRIPTION
**Pull request by hanspagel**
By popular demand, an official Docker image to run `@scalar/api-reference`

This is for people who interact more with Docker/Kubernetes than with HTML/JS/Node. Easily added to any existing Docker stack, just requires an OpenAPI URL.

**Docker**

```bash
docker run -p 1234:80 \
  -e OPENAPI_DOCUMENT_URL=https://cdn.jsdelivr.net/npm/@scalar/galaxy/dist/latest.json \
  -e API_REFERENCE_CONFIG='{"theme":"purple"}' \
  scalarapi/api-reference:latest
```

**Docker Compose**

```yaml
version: '3.8'

services:
 web:
   image: scalarapi/api-reference:latest
   ports:
     - '1234:80'
   environment:
     - OPENAPI_DOCUMENT_URL=https://cdn.jsdelivr.net/npm/@scalar/galaxy/dist/latest.json
     - API_REFERENCE_CONFIG={"theme":"purple"}
```

## Release Notes

- **New Features**
  - Introduced a Docker image for the Scalar API Reference, enhancing deployment capabilities.
  - Added a new Docker Compose configuration for easier setup and development.
  - Updated README with detailed usage instructions for the Docker container.

- **Documentation**
  - Added new sections in the README for Docker usage and configuration options.
  - Included badges for Docker pulls and Discord community engagement in the README.

- **Chores**
  - Updated workspace configuration to reflect new package structure.

### CODE REVIEW

**Comment by hanspagel** [comment]
@coderabbit review

**Comment by hanspagel** [comment]
@coderabbitai review

**Comment by hanspagel** [comment]
@CodeRabbit review

**Comment by hanspagel** [comment]
@coderabbitai review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,6 +35,71 @@ jobs:
...
+        uses: docker/build-push-action@2cdde995de11925a030ce8070c3d77a52ffcf1c0
+        with:
+          context: packages/api-reference/docker/src
+          file: packages/api-reference/docker/src/Dockerfile
+          push: false
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            scalarapi/api-reference:latest
+            scalarapi/api-reference:${{ steps.package-version.outputs.VERSION }}
+      - if: startsWith(github.event.head_commit.message, 'RELEASING:')
```
Comment (tmastrom) [inline .github/workflows/ci.yml:62]: Only runs on release 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,6 +35,71 @@ jobs:
...
+              - packages/api-reference/docker/package.json
+
+      - if: steps.changed-files.outputs.api_reference_docker_any_changed == 'true'
+        name: Log in to DockerHub
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc
+        with:
+          # Username of your Docker account
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          # https://app.docker.com/settings/personal-access-tokens
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (tmastrom) [inline .github/workflows/ci.yml:78]: Are these added to the repo?

File: packages/api-reference/docker/src/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Use the official Caddy image as a parent image
+FROM caddy:2-alpine
```
Comment (tmastrom) [inline packages/api-reference/docker/src/Dockerfile:2]: I hope caddy alpine doesn't suffer from the same issues as node alpine 😅

**Review by tmastrom — APPROVED** [review]
Looks good! Works well using the CDN for now

File: packages/api-reference/docker/src/Dockerfile (hunk 3 above)
Comment (geoffgscott) [reply packages/api-reference/docker/src/Dockerfile:2]: It does. Alpine linux is very unusual.

**Pull request merged**